Add missing types

sandrokeil opened the pull request:
**Don't merge**

sandrokeil committed:
Add missing types

codeliner commented on src/BoundedContextConnectionMap.php:
```diff
@@ -0,0 +1,210 @@
+        return new self();
+    }
+
+    public static function fromFeatureConnections(BoundedContextConnection ...$boundedContextConnections): BoundedContextConnectionMap
```
fromBoundedContextConnections, or?

sandrokeil replied:
Was removed

sandrokeil commented on src/EventSourcingAnalyzer.php:
```diff
@@ -16,5 +16,19 @@ public function commandMap(): VertexMap;
 
     public function eventMap(): VertexMap;
 
-    public function aggregateMap(): AggregateConnectionMap;
```
Is replaced by `\EventEngine\InspectioGraph\Analyzer\AggregateConnectionAnalyzer::aggregateConnectionMap()`.

sandrokeil committed:
Add vertices method

sandrokeil committed:
Connection maps need id as index, especially aggregate connection

- This is necessary to determine which command (source) produces which events (targets) for aggregates in a feature / bounded context

sandrokeil committed:
Add aggregateByDocument() method

sandrokeil committed:
Policy acts only as process manager

- supports event -> command
- supports event -> external system

sandrokeil committed:
Merge already existing aggregate connection

- needed to build a graph step by step from different stickies

sandrokeil committed:
Allow to add multiple vertexes at once

sandrokeil committed:
Merge already existing feature connection

- needed to build a graph step by step

sandrokeil committed:
Fix add same event twice

sandrokeil committed:
Fix reset internal pointer after clone

sandrokeil committed:
Optimize with methods

sandrokeil committed:
Introduce merge() method to update vertices with latest version

sandrokeil committed:
Add vertex connection map

sandrokeil committed:
Set minimum PHP version to 7.4

sandrokeil committed:
Introduce common vertex connection classes to build up the graph

- This simplifies the determination of the connections between vertices

sandrokeil committed:
Rename method vertexIdentities to vertices

sandrokeil committed:
Add more filter methods

sandrokeil committed:
Add support to search in graph for specific vertex

sandrokeil committed:
Add role type support

sandrokeil committed:
Add clearGraph() method

sandrokeil committed:
Add filterByNameAndType()

sandrokeil committed:
Add interface to resolve metadata references in graph analyzer

sandrokeil committed:
Fix wrong remaining steps calculation

sandrokeil committed:
Merge pull request #4 from event-engine/feature/simplify-connections

Introduce common vertex connection classes to build up the graph
